site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -0,0 +1,297 @@
+                    },
+                    name: name
+                }
+                return (
+                    <div {...getRootProps()}>
+                        <input {...inpProps} />
+                        <label className="helptext mt-0">
+                            {`Optional. Max attachments size:`} {humanReadableBytes(maxFileSize, false)}
+                        </label>
+                        {currentFiles.length > 0
+                            &&
+                            <FileTable
+                                handleDelete={handleDelete}
+                                filesList={currentFiles}
+                            />
+                        }
+                    </div>
+                )
+            }}
+        </Dropzone>
+    );
+};
+
+FileUploader.propTypes = {
+    dropzoneParams: PropTypes.object.isRequired,
+    maxFileSize: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    currentFiles: PropTypes.array.isRequired
+};
+
+class SupportForm extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            files: []
+        };
+    }
+
+    deserializeFieldErrors = (errors) => {
+        /**
+         * Deserialises field errors from the API to be consumed by the front-end.
+         * The output's format works with Formik.
+         */
+        let deserializedErrors = {};
+        if (Array.isArray(errors)) {
+            for (const e of errors) {
+                if (
+                    Object.prototype.hasOwnProperty.call(e, "field") &&
+                    Object.prototype.hasOwnProperty.call(e, "messages")
+                ) {
+                    set(deserializedErrors, e.field, e.messages.join(" "));
+                }
+            }
+        }
+        return deserializedErrors;
+    };
+
+    onSubmit = async (values, formikBag) => {
+        const { apiEndpoint } = this.props;
+        const formData = formikToFormData(values);
+        try {
+            const response = await http.post(apiEndpoint, formData)
+            window.location = response.request.responseURL;
+        } catch (error) {
+            let errorMessage = error.message;
+
+            // API errors need to be deserialised to highlight fields.
+            const apiResponse = error?.response?.data;
+            if (apiResponse) {
+                const apiErrors = apiResponse.errors || [];
+                const deserializedErrors = this.deserializeFieldErrors(apiErrors);
+                
+                // Highlight errors using formik
+                formikBag.setErrors(deserializedErrors);
+                errorMessage = apiResponse.message || errorMessage;
+            }
+
+            // TODO show error notification
+            console.error(errorMessage);
+        }
+    }
+
+    updateFiles = (files, formikSetFieldValue) => {
+        formikSetFieldValue("files", files)
+        this.setState({
+            files: files
```

Comment:
why do we need to keep local state for files?

Reply:
I was using it to render the list of currently accepted files.

However I realized that I can use `Formik.values` for that, therefore I am simplifying the component.